griptape/tools/file_manager/tool.py

```diff
@@ -93,9 +93,16 @@ def save_memory_artifacts_to_disk(self, params: dict) -> ErrorArtifact | InfoArt
 
         for artifact in list_artifact.value:
             formatted_file_name = f"{artifact.name}-{file_name}" if len(list_artifact) > 1 else file_name
-            result = self.file_manager_driver.save_file(os.path.join(dir_name, formatted_file_name), artifact.value)
-            if isinstance(result, ErrorArtifact):
-                return result
+            try:
+                self.file_manager_driver.save_file(os.path.join(dir_name, formatted_file_name), artifact.value)
+            except FileNotFoundError:
+                return ErrorArtifact("Path not found")
+            except IsADirectoryError:
+                return ErrorArtifact("Path is a directory")
+            except NotADirectoryError:
+                return ErrorArtifact("Not a directory")
+            except Exception as e:
+                return ErrorArtifact(f"Failed to load file: {str(e)}")
```

Comment:
should this just be the one `except` statement?

Reply:
I don't think so. We want to convey each of the exceptions to the LLM in a slightly different way. These are copied directly from the Driver that previously threw them.